### PR TITLE
Add translation options for comments and tweak UI.

### PR DESCRIPTION
**Pull request by dtjanaka**

### CODE REVIEW

File: portfolio/src/main/webapp/miscPage.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+/**
+ * Change picture for either slideshow.
+ */
+function changeSlide(slideshowNum, direction) {
+  const show = [
+    ['IMG_8732.jpg', 'IMG_8795.jpg', 'IMG_8819.jpg'],
+    ['IMG_2715.jpg', 'IMG_2718.jpg', 'IMG_2722.jpg'],
+  ];
+  curShowIdxs[slideshowNum] =
+    (((curShowIdxs[slideshowNum] + direction) % 3) + 3) % 3;
```
Comment (snazzorama) [inline portfolio/src/main/webapp/miscPage.js]: Isn't ((x % 3) + 3) % 3 equivalent to just x % 3?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -185,11 +193,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
     int maxComments = 0;
     for (Entity entity : storedComments.asIterable()) {
       String name = (String)entity.getProperty("name");
       String comment = (String)entity.getProperty("comment");
+      if (!newLang.equals("en")) {
+        try {
+            Translation translation =
+                translate.translate(comment, Translate.TranslateOption.targetLanguage(newLang));
+            comment = translation.getTranslatedText();
+        } catch(Exception e) {
```
Comment (snazzorama) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: space between "catch" and "("

File: portfolio/src/main/webapp/comments.html [hunk 3]
```diff
@@ -76,6 +76,17 @@ <h1>Comments</h1>
             <option value="dsc" selected>Date: new to old</option>
             <option value="asc">Date: old to new</option>
           </select>
+          <label for="lang">Language:</label>
```
Comment (snazzorama) [inline portfolio/src/main/webapp/comments.html]: As I understand it, the "for" attribute is supposed to match up with the ID of an input. But that doesn't seem to be the case here. Why do you have 'for="lang"'?

File: portfolio/src/main/webapp/miscPage.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+/**
+ * Creates a map and adds it to the page.
+ */
+function createMap() {
+  const map = new google.maps.Map(document.getElementById('map'), {
+    //center: {lat: 37.412177, lng: -121.959926},
+    center: { lat: 37.39, lng: -121.945 },
+    zoom: 13,
+  });
+  const GRTStartMarker = new google.maps.Marker({
```
Comment (snazzorama) [inline portfolio/src/main/webapp/miscPage.js]: Acronyms should be put into camelcase as if they were a word. So "grtStartMarker".

https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined

File: portfolio/src/main/webapp/miscPage.js (hunk 1 above)
Comment (dtjanaka) [reply portfolio/src/main/webapp/miscPage.js]: For -1 % 3, JS returns -1 instead of 2.

Comment (robotsaurus) [reply portfolio/src/main/webapp/miscPage.js]: Fun fact: different languages will return different values for modulo of negative numbers....

If there isn't a way to simplify this then I would suggest adding a comment on what this is trying to do because it is definitely hard to parse.  If someone had to modify this then they would definitely probably get it wrong.

You could also consider breaking it apart as:

curShowIdxs[slideshowNum] = (curShowIdxs[slideshowNum] + direction) % 3;
// We need to handle negative numbers as -1 % 3 is -1.
if (curShowIdxs[slideshowNum] < 0) {
  curShowIdxs[slideshowNum] += 3;
}

**Review by robotsaurus — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/miscPage.js (hunk 1 above)
Comment (snazzorama) [reply portfolio/src/main/webapp/miscPage.js]: Yeah, thanks for the explanation. JavaScript is a language full of hidden delights like this. 😑

Looking online, it seems like your approach is idiomatic in JavaScript. As Mike said, a comment would help, or you could put that math in your own mod() function with a comment there.

**Comment by dtjanaka** [comment]
PTAL

**Review by snazzorama — APPROVED** [review]
